### PR TITLE
Sign tool fixes

### PR DESCRIPTION
**Pull request by tmat**
Implements a couple of missing features:
- support for TargetFrameworkAttribute defined in the assembly that uses it
- deduplication of artifacts to sign based on content hash

Also adds more tests and cleans up the code base.

### CODE REVIEW

**Comment by tmat** [comment]
@JohnTortugo PTAL

File: src/Microsoft.DotNet.SignTool/src/ByteSequenceComparer.cs [hunk 1]
```diff
@@ -0,0 +1,122 @@
...
+            {
+                return ReferenceEquals(left, right);
+            }
+
+            if (ReferenceEquals(left, right) && leftStart == rightStart)
+            {
+                return true;
+            }
+
+            for (var i = 0; i < length; i++)
```
Comment (jaredpar) [inline src/Microsoft.DotNet.SignTool/src/ByteSequenceComparer.cs]: This needs to guard against the index going past the bounds of the array.

Comment (tmat) [reply src/Microsoft.DotNet.SignTool/src/ByteSequenceComparer.cs]: Yes, it should. Copied this from Roslyn... Will delete the method here, it's not used.

---
In reply to: [212344527](https://github.com/dotnet/arcade/pull/518#discussion_r212344527) [](ancestors = 212344527)

Comment (jaredpar) [reply src/Microsoft.DotNet.SignTool/src/ByteSequenceComparer.cs]: Where in roslyn? We should fix.

Comment (tmat) [reply src/Microsoft.DotNet.SignTool/src/ByteSequenceComparer.cs]: https://github.com/dotnet/roslyn/blob/fab7134296816fc80019c60b0f5bef7400cf23ea/src/Compilers/Core/Portable/Collections/ByteSequenceComparer.cs#L44

**Pull request merged**